src/iop/denoiseprofile.c

```diff
@@ -314,6 +314,29 @@ typedef struct dt_iop_denoiseprofile_global_data_t
 
 static dt_noiseprofile_t dt_iop_denoiseprofile_get_auto_profile(dt_iop_module_t *self);
 
+#ifdef DEBUG_SCALES
+static void debug_dump_PFM(const dt_dev_pixelpipe_iop_t *const piece, const char *const namespec,
+                           const float* const restrict buf, const int width, const int height, const int scale)
+{
+  if((piece->pipe->type & DT_DEV_PIXELPIPE_PREVIEW) != DT_DEV_PIXELPIPE_PREVIEW)
+  {
+    char filename[512];
+    snprintf(filename, sizeof(filename), namespec, scale);
+    FILE *f = g_fopen(filename, "wb");
+    if (f)
+    {
+      fprintf(f, "PF\n%d %d\n-1.0\n", width, height);
+      const size_t n = (size_t)width * height;
+      for(int k=0; k<n; k++)
```

Comment:
since k can't be below 0, maybe size_t it?

Reply:
Missed that one.